### PR TITLE
Change team mode on exercise import.

### PR DESCRIPTION
**Pull request by sleiss**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Client: I documented the TypeScript code using JSDoc style.

### Motivation and Context
See #2265: The team mode of exercises can not be changed during import, although there is no reason why it's not technically possible.

### Description
The check in the client that determined whether the team mode can be changed was faulty: It should only disable changes when editing the exercise, not when importing.
The server already supported this correctly, but I added JUnit tests to verify this.

### Steps for Testing
1. Create a (programming) exercise in individual mode
2. Import that exercise into a new exercise but change to team mode
3. Verify that everything works as expected: Teams can be created, students can be assigned to teams, etc.
4. Repeat steps 1-3 but swap individual & team mode (start with team mode and import to individual mode).
5. Make sure that you can not change the exercise mode when editing the exercises.

### CODE REVIEW

**Review by tiffca — APPROVED** [review]
Tested the scenarios and all worked as expected. No issues importing, participating in team/individual exercises via online/offline mode, and confirmed cannot edit the exercise team/individual mode after import. Good job!

File: src/test/java/de/tum/in/www1/artemis/util/ProgrammingExerciseTestService.java [hunk 1]
```diff
@@ -307,6 +309,85 @@ public void importExercise_created(ProgrammingLanguage programmingLanguage) thro
                 .containsExactlyInAnyOrderElementsOf(sourceExercise.getExerciseHints());
     }
 
+    // TEST
```
Comment (stefanwaldhauser) [inline src/test/java/de/tum/in/www1/artemis/util/ProgrammingExerciseTestService.java:312]: Remove?

**Review by stefanwaldhauser — APPROVED** [review]
Tested it for programming exercises and everything workes on TS3. Gj

**Review by FuchsDominik — APPROVED** [review]
Works like a charm! Everything worked, from being able to switch to Individual <-> Teams and other way around, adding teams + members and participating in the exercise. Editing afterwards was also not possible.

**Pull request merged**